parso/cache.py

```diff
@@ -187,6 +210,37 @@ def clear_cache(cache_path=None):
     parser_cache.clear()
 
 
+def clear_inactive_cache(
+    cache_path=None,
+    inactivity_threshold=_CACHED_FILE_MAXIMUM_SURVIVAL,
+):
+    cache_path = _get_default_cache_path()
```

Comment:
I think we should iterate through all the folders in `~/.cache/parso` instead of just the specific version. This also means that we might need to change something about `_get_cache_clear_lock` and that it should be "global".

Reply:
The reasoning for this is mostly that people will upgrade Python versions and in that case the old caches will just stick around forever, which makes no sense. For example in my case I have the following folders:

```
CPython-27-30  CPython-33-30  CPython-34-33  CPython-35-33  CPython-36-33  CPython-37-33
CPython-27-31  CPython-34-30  CPython-35-30  CPython-36-30  CPython-37-30  CPython-38-31
CPython-27-32  CPython-34-31  CPython-35-31  CPython-36-31  CPython-37-31  CPython-38-32
CPython-27-33  CPython-34-32  CPython-35-32  CPython-36-32  CPython-37-32  CPython-38-33
```

The whole thing uses 3.5 GB. :)